### PR TITLE
UIU-2873: allow to run query without grantedToIds

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIU-2873](https://issues.folio.org/browse/UIU-2873) - allow to run query with empty grantedToIds
The reason to allow run query when the user removes all assigned users the list isn't updated due to `grantedToIds` length equal to 0.
## Approach

https://github.com/folio-org/ui-users/assets/116072773/abf5cef7-fbb1-42e0-8cad-cc8906578712
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ const useAssignedUsers = ({ grantedToIds = [], permissionSetId, tenantId }, opti
       }));
     },
     {
-      enabled: Boolean(grantedToIds.length && permissionSetId),
+      enabled: Boolean(permissionSetId),
```
Comment (NikitaSedyx) [inline src/components/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.js:82]: if you enable query when array is empty, do no load data in querFn when array is empty - there is no sense to load nothing

Comment (alisher-epam) [reply src/components/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.js]: Agree on that.

File: src/components/AssignedUsers/AssignedUsersContainer.js [hunk 2]
```diff
@@ -109,7 +109,7 @@ const AssignedUsersContainer = ({ permissionSetId, expanded, onToggle, tenantId
         : (
           <AssignedUsersList
             isFetching={isFetching || isMutationLoading}
-            users={users}
+            users={grantedTo.length ? users : []}
```
Comment (NikitaSedyx) [inline src/components/AssignedUsers/AssignedUsersContainer.js:112]: looks strange, you still can enable query, but you can return empty array before loading data when array is empty

**Review by BogdanDenis — APPROVED** [review]
Please also update CHANGELOG

File: src/components/AssignedUsers/AssignedUsersContainer.js (hunk 2 above)
Comment (alisher-epam) [reply src/components/AssignedUsers/AssignedUsersContainer.js:112]: @NikitaSedyx  What do you suggest here?

Comment (NikitaSedyx) [reply src/components/AssignedUsers/AssignedUsersContainer.js:112]: > you still can enable query, but you can return empty array before loading data when array is empty

**Review by Terala-Priyanka — APPROVED** [review]

**Pull request merged**